File: src/main/java/ro/fasttrackit/homework/ItemValidator.java
package ro.fasttrackit.homework;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class ItemValidator {
    private void verifyCategory(Category category) {
        verifyNotNull(category);
    }

    private void verifyPrice(int price) {
        if (price <= 0) throw new IllegalArgumentException("invalid price");
    }

    private void verifyName(String name) {
        verifyNotNull(name);
        Pattern pattern = Pattern.compile("^[ A-za-z0-9'\\-]+$");
        Matcher matcher = pattern.matcher(name);
        if (!matcher.matches()) throw new IllegalArgumentException("invalid name");
    }

    private <T> void verifyNotNull(T t) {
        if (t == null || t.equals("")) throw new IllegalArgumentException();
    }

    public void verify(String name, int price, Category category) {
        verifyName(name);
        verifyPrice(price);
        verifyCategory(category);
    }
}

File: src/main/java/ro/fasttrackit/khomework/ItemValidator.kt
package ro.fasttrackit.khomework

import java.util.regex.Pattern

class ItemValidator {

    private fun <T> verifyNotNull(t: T) {
        if (t == null || t == "") throw Exception()
    }

    private fun verifyName(name: String) {
        verifyNotNull(name)
        val pattern = Pattern.compile("^[ A-za-z0-9'\\-]+$")
        val matcher = pattern.matcher(name)
        if (!matcher.matches()) throw Exception()
    }

    private fun verifyPrice(price: Int) {
        if (price <= 0) throw Exception()
    }

    private fun verifyCategory(category: Category) {
        verifyNotNull(category)
    }

    fun verify(name: String, price: Int, category: Category) {
        verifyName(name)
        verifyPrice(price)
        verifyCategory(category)
    }
}